File: courses/codes/LightSol/Light.py
class Light:
    RED = "red"
    BLUE = "blue"
    GREEN = "green"
    WHITE = "white"

    def __init__(self, o=True, b=False, c=WHITE):
        self.burnt_out = b
        self.on = not b and o
        # self.color = c
        self.set_color(c)

    def __str__(self):
        return ("on " if self.on else "off ") + self.color + ("\t" if self.burnt_out else "\tnot ") + "burnt out"

    def burn_out(self):
        self.burnt_out = True
        self.on = False

    def flip(self):
        self.on = not self.on and not self.burnt_out

    def get_color(self):
        return self.color

    def set_color(self, c):
        self.color = c.lower()
        if self.color != Light.RED and self.color != Light.BLUE and self.color != Light.GREEN:
            self.color = Light.WHITE

    def is_on(self):
        return self.on

File: courses/codes/python-class/LightSol/Light.java
import java.util.Locale;

public class Light {
    public static String RED = "red";
    public static String BLUE = "blue";
    public static String GREEN = "green";
    public static String WHITE = "white";

    boolean burntOut;
    boolean on;
    String color;

    public Light() {
        this(true, false, WHITE);
//    	this(true, false, "white");
    }

    public Light(boolean o, boolean b, String c) {
        this.burntOut = b;
        this.on = !b && o;
//        this.color = c;
        this.setColor(c);
    }

    public void burnOut() {
        this.burntOut = true;
        this.on = false;
    }

    public void flip() {
        this.on = !this.on && !this.burntOut;
    }

    public String getColor() {
        return this.color;
    }

    public void setColor(String c) {
        this.color = c.toLowerCase();
        if (this.color != RED && this.color != BLUE && this.color != GREEN) {
        	this.color = WHITE;
        }
    }

    public boolean isOn() {
        return this.on;
    }

    @Override
    public String toString() {
        return (this.on ? "on " : "off ") + this.color + (this.burntOut ? "\t" : "\tnot ") + "burnt out";
    }
}


